services/task-svc/internal/patient/patient.go

```diff
@@ -361,6 +368,8 @@ func (ServiceServer) GetPatientDetails(ctx context.Context, req *pb.GetPatientDe
 		}
 	})
 
+	tracking.AddPatientToRecentActivity(ctx, id.String())
```

Comment:
Add this patient to recent activity in order to suggest a shorter patient list is pretty complicated, as this function will be requested pretty often without any intention of operating or managing that patient.

So I would argue that this is no necessarily the best place to detect recent activity, but that's discussable (we could try this).
But definitely add `AssignBed` and `UnassignBed` to the "detection list".